pkgs/development/python-modules/mmcv/default.nix

```
@@ -0,0 +1,67 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, python3
+, pythonOlder
+, requests
+, cudaSupport ? false
+, cudatoolkit
+, torch-cuda-arch-list ? "Kepler"
+, torch
+, torchvision
+, torch-bin
+, torchvision-bin
+, opencv4
+, yapf
+, packaging
+, pillow
+, addict
+, ninja
+, which
+}:
+
+buildPythonPackage rec {
+  pname = "mmcv";
+  version = "1.7.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "open-mmlab";
+    repo = pname;
+    rev = "v1.7.0";
+    hash = "sha256-EVu6D6rTeebTKFCMNIbgQpvBS52TKk3vy2ReReJ9VQE=";
+  };
+
+  torch-deps = if cudaSupport then [ torch-bin torchvision-bin ] else [ torch torchvision ];
+  nativeBuildInputs = [ ninja which ]
+    ++ lib.lists.optionals cudaSupport [ cudatoolkit.lib cudatoolkit.out cudatoolkit.cc ] ++ torch-deps;
+
+  propagatedBuildInputs = [
+    opencv4
+    yapf
+    packaging
+    pillow
+    addict
+  ] ++ torch-deps;
+
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "mmcv"
+  ];
+
+  MMCV_WITH_OPS = 1;
+  FORCE_CUDA = cudaSupport;
+  TORCH_CUDA_ARCH_LIST = torch-cuda-arch-list;
+  LDFLAGS = if cudaSupport then "-L${cudatoolkit.lib}/lib" else "";
+
+
+  meta = with lib; {
+    description = "MMCV is a foundational library for computer vision research";
+    homepage = "https://github.com/open-mmlab/mmcv";
```

Allows tools (e.g., `nix-update`) to add the changelog to the commit message while updating.

```suggestion
    homepage = "https://github.com/open-mmlab/mmcv";
    changelog = "https://github.com/open-mmlab/mmcv/releases/tag/v${version}";
```